example/fine_advection/parthenon_app_inputs.cpp

```diff
@@ -0,0 +1,119 @@
+//========================================================================================
+// (C) (or copyright) 2020-2021. Triad National Security, LLC. All rights reserved.
+//
+// This program was produced under U.S. Government contract 89233218CNA000001 for Los
+// Alamos National Laboratory (LANL), which is operated by Triad National Security, LLC
+// for the U.S. Department of Energy/National Nuclear Security Administration. All rights
+// in the program are reserved by Triad National Security, LLC, and the U.S. Department
+// of Energy/National Nuclear Security Administration. The Government is granted for
+// itself and others acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+// license in this material to reproduce, prepare derivative works, distribute copies to
+// the public, perform publicly and display publicly, and to permit others to do so.
+//========================================================================================
+
+#include <sstream>
+#include <string>
+
+#include <parthenon/package.hpp>
+
+#include "advection_driver.hpp"
+#include "advection_package.hpp"
+#include "config.hpp"
+#include "defs.hpp"
+#include "interface/variable_pack.hpp"
+#include "kokkos_abstraction.hpp"
+#include "parameter_input.hpp"
+#include "utils/error_checking.hpp"
+
+using namespace parthenon::package::prelude;
+using namespace parthenon;
+
+// *************************************************//
+// redefine some weakly linked parthenon functions *//
+// *************************************************//
+
+namespace advection_example {
+
+void ProblemGenerator(MeshBlock *pmb, ParameterInput *pin) {
+  using parthenon::MetadataFlag;
+
+  auto &data = pmb->meshblock_data.Get();
+
+  auto pkg = pmb->packages.Get("advection_package");
+  const auto &profile = pkg->Param<std::string>("profile");
+
+  auto cellbounds = pmb->cellbounds;
+  IndexRange ib = cellbounds.GetBoundsI(IndexDomain::interior);
+  IndexRange jb = cellbounds.GetBoundsJ(IndexDomain::interior);
+  IndexRange kb = cellbounds.GetBoundsK(IndexDomain::interior);
+
+  auto coords = pmb->coords;
+
+  using scalar = advection_package::Conserved::scalar;
+  using scalar_fine = advection_package::Conserved::scalar_fine;
+  static auto desc = parthenon::MakePackDescriptor<scalar, scalar_fine>(data.get());
+  auto pack = desc.GetPack(data.get());
+
+  int profile_type;
+  if (profile == "wave") profile_type = 0;
+  if (profile == "smooth_gaussian") profile_type = 1;
+  if (profile == "hard_sphere") profile_type = 2;
+  if (profile == "block") profile_type = 3;
+  Real amp = 1.0;
+
+  const int b = 0;
+  const int ndim = pmb->pmy_mesh->ndim;
+  const int nghost = parthenon::Globals::nghost;
+  pmb->par_for(
```

Comment:
should we avoid using `pmb->` now that overloads exit that don't need the loop pattern and exec space specified? I feel like `pmb->par_for` should eventually go away.

Reply:
oh, yeah. I copied the old advection example and updated it. I thought I removed all `pmb->par_for`s but apparently not.

Reply:
this is fixed in a subsequent PR